packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -361,6 +368,17 @@ export default class DatagridController {
         });
     }
 
+    applyRowsSelection () {
+        if (this.keepSelectedRows) {
+            this.currentSelectedRows = this.displayedRows.map(row => !!find(this.selectedRows, selectedRow => isEqual(selectedRow, row)));
+            this.selectAllRows = this.currentSelectedRows.filter(isSelected => isSelected).length === this.displayedRows.length;
+        } else {
+            this.selectedRows = [];
+            this.currentSelectedRows = this.currentSelectedRows.map(() => false);
```

Comment:
Both `currentSelectedRows` and `selectAllRows` are always updated at the same time. This is code smell